iio/iio.c

```diff
@@ -736,49 +796,70 @@ static int iio_write_attr(struct iiod_ctx *ctx, const char *device,
 			  struct iiod_attr *attr, char *buf, uint32_t len)
 {
 	struct iio_dev_priv	*dev;
+	struct iio_trig_priv *trig_dev;
 	struct attr_fun_params	params;
 	struct iio_attribute	*attributes;
 	struct iio_ch_info ch_info;
 	struct iio_channel *ch = NULL;
 	int8_t ch_out;
 
 	dev = get_iio_device(ctx->instance, device);
-	if (!dev)
-		return -ENODEV;
 
-	if (attr->type == IIO_ATTR_TYPE_DEBUG &&
-	    strcmp(attr->name, REG_ACCESS_ATTRIBUTE) == 0) {
-		if (dev->dev_descriptor->debug_reg_write)
-			return debug_reg_write(dev, buf, len);
-		else
-			return -ENOENT;
-	}
+	/* If IIO device with given name is found, handle writing of attributes */
+	if (dev) {
 
-	if (attr->channel) {
-		ch_out = attr->type == IIO_ATTR_TYPE_CH_OUT ? 1 : 0;
-		ch = iio_get_channel(attr->channel, dev->dev_descriptor,
-				     ch_out);
-		if (!ch)
-			return -ENOENT;
+		if (attr->type == IIO_ATTR_TYPE_DEBUG &&
+		    strcmp(attr->name, REG_ACCESS_ATTRIBUTE) == 0) {
+			if (dev->dev_descriptor->debug_reg_write)
+				return debug_reg_write(dev, buf, len);
+			else
+				return -ENOENT;
+		}
 
-		ch_info.ch_out = ch_out;
-		ch_info.ch_num = ch->channel;
-		ch_info.type = ch->ch_type;
-		ch_info.differential = ch->diferential;
-		ch_info.address = ch->address;
-		params.ch_info = &ch_info;
-	} else {
-		params.ch_info = NULL;
-	}
+		if (attr->channel) {
+			ch_out = attr->type == IIO_ATTR_TYPE_CH_OUT ? 1 : 0;
+			ch = iio_get_channel(attr->channel, dev->dev_descriptor,
+					     ch_out);
+			if (!ch)
+				return -ENOENT;
+
+			ch_info.ch_out = ch_out;
+			ch_info.ch_num = ch->channel;
+			ch_info.type = ch->ch_type;
+			ch_info.differential = ch->diferential;
+			ch_info.address = ch->address;
+			params.ch_info = &ch_info;
+		} else {
+			params.ch_info = NULL;
+		}
 
-	params.buf = (char *)buf;
-	params.len = len;
-	params.dev_instance = dev->dev_instance;
-	attributes = get_attributes(attr->type, dev, ch);
-	if (!strcmp(attr->name, ""))
-		return iio_write_all_attr(&params, attributes);
-	else
-		return iio_rd_wr_attribute(&params, attributes, attr->name, 1);
+		params.buf = (char *)buf;
+		params.len = len;
+		params.dev_instance = dev->dev_instance;
+		attributes = get_attributes(attr->type, dev, ch);
+		if (!strcmp(attr->name, ""))
+			return iio_write_all_attr(&params, attributes);
+		else
```

Comment:
redundant else

Reply:
could you please elaborate? this else is actually being used to call iio_rd_wr_attribute to write to a single attribute.